src/subdomains/supporting/address-pool/deposit/deposit.service.ts

```diff
@@ -97,18 +99,38 @@ export class DepositService {
     for (let i = 0; i < count; i++) {
       const accountIndex = nextDepositIndex + i;
 
-      const wallet = EvmUtil.createWallet(Config.blockchain.evm.walletAccount(accountIndex));
-      const deposit = Deposit.create(wallet.address, applicableChains, accountIndex);
-      await this.depositRepo.save(deposit);
-
-      addresses.push(deposit.address);
+      addresses.push(
+        await this.saveEvmDeposit(Config.blockchain.evm.walletAccount(accountIndex), applicableChains, accountIndex),
+      );
     }
 
+    addresses.push(this.createPaymentAddress(0));
+
     for (const chain of applicableChains) {
       await this.alchemyWebhookService.createAddressWebhook({ blockchain: chain, addresses: addresses });
     }
   }
 
+  private createPaymentAddress(accountIndex: number): string {
+    return EvmUtil.createWallet({ seed: Config.payment.evmSeed, index: accountIndex }).address;
+  }
+
+  private async saveEvmDeposit(
+    walletAccount: WalletAccount,
+    applicableChains: Blockchain[],
+    accountIndex: number,
+  ): Promise<string> {
+    const wallet = EvmUtil.createWallet(walletAccount);
+
+    const deposit = await this.depositRepo.findOneBy({ address: wallet.address });
+
+    if (!deposit) {
+      await this.depositRepo.save(Deposit.create(wallet.address, applicableChains, accountIndex));
```

Comment:
This check is not necessary anymore

Reply:
I have restored it to its previous state